.gitignore

```diff
@@ -1,27 +1,106 @@
+# Byte-compiled / optimized / DLL files
+__pycache__/
+*.py[cod]
+*$py.class
+
+# C extensions
+*.so
+
+# I don't konw.
 _trial_temp
-build
-build-pypy
-MANIFEST
-dist
 apidocs
 *,cover
 .testrepository
-*.pyc
-*.pyd
-*.pyo
-*.so
 *~
-*.swp
-*.swh
-*.swn
-*.swo
+*.sw[phno]
 docs/tutorial/index.html
 dulwich.egg-info/
+docs/api/*.txt
+dulwich.dist-info
+
+# Unit test / coverage reports
+htmlcov/
 .tox/
+.coverage
+.coverage.*
+.cache
+nosetests.xml
+coverage.xml
+*.cover
+.hypothesis/
+.pytest_cache/
 .idea/
 .coverage
-htmlcov/
-docs/api/*.txt
+
+# Distribution / packaging
+.Python
+build/
+develop-eggs/
+dist/
+downloads/
+eggs/
+.eggs/
+lib/
+lib64/
+parts/
+sdist/
+var/
+wheels/
+*.egg-info/
+.installed.cfg
+*.egg
+MANIFEST
```

Comment:
*.egg *.egg-info, dist and build are the only files that are relevant here. Dulwich doesn't create any of the other directories listed here.

Reply:
You still have C language to construct and deploy, so keep the others, unless you don’t have C language in this project, but as far as I know